idaplugin/rematch/collectors/identity_hash.py

```diff
@@ -0,0 +1,34 @@
+import idautils
+import idc
+
+from .vector import Vector
+
+
+class IdentityHashVector(Vector):
+  type = 'identity_hash'
+  type_version = 0
+
+  # The Keleven
+  # http://movies.stackexchange.com/q/11495
+  keleven = 17391172068829961267
+
+  def __init__(self, **kwargs):
+    super(IdentityHashVector, self).__init__(**kwargs)
+    self.hash = self.keleven
+
+  def _cycle(self, b):
+    self.hash |= 5
+    self.hash ^= b
+    self.hash *= self.hash
+    self.hash ^= (self.hash >> 32)
+    self.hash &= 0xffffffffffffffff
+
+  def _data(self):
+    for ea in idautils.FuncItems(self.offset):
+      self._cycle(idc.Byte(ea))
+      # skip additional bytes of any instruction that contains an offset in it
+      if idautils.CodeRefsFrom(ea, True) or idautils.DataRefsFrom(ea):
```

Comment:
According to [documentation](https://www.hex-rays.com/products/ida/support/sdkdoc/classfunc__item__iterator__t.html), `idautils.FuncItems` should return  *function* code and data items. Its not quite clear to me how can data be part of a function. For example, in the example you provided I would think lines 1-3 would count for a single function chunk and 5-7 will count as another function chunk, leaving the data in line 4 outside the function.

To test I took a function, converted a few instructions to data and called `FuncItems` on that function. data addresses were excluded from the yielded items.

Are you able to reproduce such a scenario?